pkg/extensions/tbtc/tbtc.go

```diff
@@ -16,18 +19,21 @@ import (
 
 var logger = log.Logger("tbtc-extension")
 
-const maxActAttempts = 3
+const (
+	maxActAttempts           = 3
+	pastEventsLookbackBlocks = 10000
```

Comment:
This is possible but I'm afraid this can lead to some problems in the future. I'm not sure if we can safely rely on blocks numbers taken from events due to some weird things which may happen like event duplications or chain reorgs. Also, this will work for all actions but `provideRedemptionProof` which should look in the past beyond the monitoring start block. Taking those concerns in account, I think the safest and simplest way is just start from block `current - 10k`

Reply:
We can leave it like that for now. A little concerned about how much it costs to scan 10k previous blocks for operators using 3rd party Ethereum providers who charge by compute unit but we can revise in the future.

Reply:
IIRC, it's always one `eth_getLogs` call if we are talking about the number of requests.

Reply:
Yeah, though it sometimes does not matter if it's just one call but how expensive was to compute a respose for this call.